### DISCUSSION TITLE
ValueError: Invalid padding bytes.

### DISCUSSION THREAD

**Discussion by paulhan93** [JECSand/yahoofinancials#121] [General]
Hello, is anyone else having issues when trying to run YahooFinance methods? I tried running "get_current_price()" and "get_pe_ratio()" and got "ValueError: Invalid padding bytes." It was working fine yesterday.

**Comment by paulhan93** [comment]
oops, just saw that I should have posted this on Issues.

**Reply by JECSand** [reply]
@paulhan93 Issue is fixed now with v1.11.

However this thread can be used as a place to discuss the recent changes with the Yahoo Finance web app. Basically there’s an ongoing cat and mouse game between Yahoo and open source developers, which is impacting yahoo finance libraries. Most of the other projects relying on the Yahoo Finance data store for fundamental financial data are still down as of right now.

Essentially Yahoo is encrypting their web app’s data store with increasingly sophisticated public decryption key management techniques. This results in open source api projects, such as yahoofinancials, having to rapidly improve their logic locating these public keys for decryption. 

Unless I post otherwise my goal is to try to have any (likely) future issues fixed within 48 hours. Luckily unless Yahoo hides their data behind a pay wall, there will always be a way to decrypt it. That being said I wouldn’t rule out that eventual possibility, which is why I’m planning on updating yahoofinancials with the logic needed to allow end users to request data using their authenticated paid yahoo accounts.
